src/common/libkvs/kvs_lookup.c

```diff
@@ -88,16 +88,16 @@ static int validate_lookup_flags (int flags, bool watch_ok)
 {
     if ((flags & FLUX_KVS_WATCH) && !watch_ok)
         return -1;
-    if (flags & FLUX_KVS_WATCH_WAITCREATE
-        && !(flags & FLUX_KVS_WATCH))
-        return -1;
-    if (flags & FLUX_KVS_WATCH_FULL
+    if ((flags & FLUX_KVS_WATCH_WAITCREATE
+         || flags & FLUX_KVS_WATCH_FULL
+         || flags & FLUX_KVS_WATCH_UNIQ)
         && !(flags & FLUX_KVS_WATCH))
```

Comment:
Would it be slightly more maintainable to 
```C
#define FLUX_KVS_WATCH_FLAGS = FLUX_KVS_WATCH_WAITCREATE | FLUX_KVS_WATCH_FULL | FLUX_KVS_WATCH_UNIQ
```
or similar, then test `if ((flags & FLUX_KVS_WATCH_FLAGS) && !(flags & FLUX_KVS_WATCH))`

New future watch flags could then be added in one place at the top of the file.

This would allow clearing of all watch flags below to be simplified as well.

As far as I'm concerned this is just a suggestion, not a requirement for the PR to go in.

Reply:
that's a good idea